src/lib/resource.js

```diff
@@ -1,51 +1,139 @@
-/* eslint-disable import/prefer-default-export */
-import httpCodes from 'http-codes';
-import axios from 'axios';
-import { auditLogger, logger } from '../logger';
-import { Resource } from '../models';
-
-const getResourceMetaDataJob = async (job) => {
-  const {
-    resourceId, resourceUrl,
-  } = job.data;
-  let res;
-  try {
-    // Use axios to get url info.
-    res = await axios.get(resourceUrl, { maxRedirects: 3 });
-
-    // Get page title.
-    const foundTitle = res.data.match(/<title[^>]*>([^<]+)<\/title>/);
-
-    // If we found a title, update the resource.
-    if (foundTitle && foundTitle.length >= 1) {
-      // Get title.
-      const titleToUpdate = foundTitle[1].trim();
-
-      // Update URL in DB.
-      const updatedCnt = await Resource.update({
-        title: titleToUpdate,
-      }, {
-        where: { url: resourceUrl },
-        individualHooks: false,
-      });
-
-      // If we don't update anything throw an error to retry.
-      if (updatedCnt[0] === 0) {
-        throw Error('Failed to update resource title.');
-      }
-    } else {
-      auditLogger.info(`Resource Queue: Warning, unable to retrieve resource metadata for resource '${resourceUrl}'.`);
-      return ({ status: httpCodes.NOT_FOUND, data: { url: resourceUrl } });
-    }
-    logger.info(`Resource Queue: Successfully retrieved resource metadata for resource '${resourceUrl}'`);
-    return ({ status: httpCodes.OK, data: { url: resourceUrl } });
-  } catch (error) {
-    auditLogger.error(`Resource Queue: Unable to retrieve title for Resource (ID: ${resourceId} URL: ${resourceUrl}), please make sure this is a valid address:`, error);
-    throw Error(error); // We must rethrow the error here to ensure the job is retried.
-  }
-};
-
-export {
-  axios,
-  getResourceMetaDataJob,
-};
+/* eslint-disable import/prefer-default-export */
+import httpCodes from 'http-codes';
+import axios from 'axios';
+import { auditLogger, logger } from '../logger';
+import { Resource } from '../models';
+
+const getPageScrapeValues = async (resourceUrl, isEclkc) => {
+  // Use axios to get url info.
+  const res = await axios.get(resourceUrl, { maxRedirects: 3 });
+
+  // Scrape Title & National Centers from ECLKC resources.
+  const foundTitle = res.data.match(/<title[^>]*>([^<]+)<\/title>/);
+  // eslint-disable-next-line max-len
+  // Find the meta tag with name="national-centers".
+  const foundNationalCenters = isEclkc ? res.data.match(/<meta[^>]*=*"national-centers"[^>]*content="([^"]+)"[^>]*>/) : null;

```

Comment:
I would suggest using the native `DOMParser` API for this, instead of regular expression matching, e.g.

```javascript
  const parser = new DOMParser();
  const doc = parser.parseFromString(response.data, "text/html");
  
  const updateObj = {};
  
  if (doc.title) {
    const titleToUpdate = doc.title.trim();
    updateObj.title = titleToUpdate;
  }
  
  if (isEclkc) {
    const nationalCentersNode = doc.querySelector("meta[name='national-centers']");
    if (nationalCentersNode && nationalCentersNode.hasAttribute("content")) {
      const nationalCentersToUpdate = nationalCentersNode.getAttribute("content").trim();
      updateObj.nationalCenters = nationalCentersToUpdate;
    }
  }
```

in my opinion this is less verbose, easier to read and more maintainable for those of us who are not regex wizards. note this is untested and may require slight modifications